core/src/main/resources/swagger/api-docs.yaml

```diff
@@ -4295,6 +4295,43 @@ paths:
             '*/*':
               schema:
                 $ref: '#/components/schemas/ErrorReport'
+
+  /api/workspaces/{workspaceNamespace}/{workspaceName}/snapshots/v2/query/{referencedSnapshotId}:
```

Comment:
I think instead of this being its own endpoint, it may be better to support `referencedSnapshotId` as one of the query params on the existing listSnapshots endpoint

Reply:
oooh hmm I like that, lemme work on that!

Reply:
Ahh I'm remembering now why I didn't do it this way.

The existing listSnapshots endpoint supports pagination. We pass through the pagination params to WSM, and WSM generates a paginated SQL statement to the db.

My current implementation of search-for-referencedId does not support pagination. Because the search logic is all in Rawls, it's a different beast and can't rely on SQL's `limit`/`offset`. So I see a few options:

1. Leave PR as-is, with a separate endpoint and no pagination
2. Implement pagination inside Rawls for the search functionality
3. Implement search-for-referencedSnapshotId inside WSM and pass through pagination to WSM

I guess option 2 is my choice, @MatthewBemis @ansingh7115 what do you think?

Reply:
blah....that's a PITA. I was really hoping that we'd get pagination for free here. I am fine with 2 or 3. I think I would prefer option 3 because it feels like a generic field filtering option would be more appropriate living in WSM....but I also think it's probably the most expedient route to go with option 2, so 👍